### PR TITLE
Update URDF examples

### PR DESCRIPTION
**Pull request by bmagyar**
Following #27

### CODE REVIEW

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 1]
```diff
@@ -38,25 +38,22 @@ Note:
       <param name="example_param_read_for_sec">2</param>
     </hardware>
     <joint name="joint1">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:41]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 2]
```diff
@@ -415,285 +414,284 @@ Note:
...
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:436]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 3]
```diff
@@ -415,285 +414,284 @@ Note:
...
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:458]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 4]
```diff
@@ -415,285 +414,284 @@ Note:
...
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>contact_information</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:481]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="contact_information"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 5]
```diff
@@ -415,285 +414,284 @@ Note:
...
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:503]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 6]
```diff
@@ -415,285 +414,284 @@ Note:
...
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:525]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 7]
```diff
@@ -415,285 +414,284 @@ Note:
...
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>contact_information</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:548]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="contact_information"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 8]
```diff
@@ -415,285 +414,284 @@ Note:
...
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:570]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 9]
```diff
@@ -415,285 +414,284 @@ Note:
...
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:592]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 10]
```diff
@@ -415,285 +414,284 @@ Note:
...
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>contact_information</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:615]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="contact_information"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 11]
```diff
@@ -415,285 +414,284 @@ Note:
...
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:638]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 12]
```diff
@@ -415,285 +414,284 @@ Note:
...
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:660]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 13]
```diff
@@ -415,285 +414,284 @@ Note:
...
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>contact_information</state_interface>
+      <state_interface>timestamp</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:683]:
```suggestion
      <command_interface name="Kp">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="Kd">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort_limit">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
      <state_interface name="coil_resistance"/>
      <state_interface name="contact_information"/>
      <state_interface name="timestamp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 14]
```diff
@@ -415,285 +414,284 @@ Note:
...
     </joint>
 
     <sensor name="sensor1">
       <classType>ros2_control_components/IMUSensor</classType>
-      <stateInterfaceType>orientation</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>acceleration</stateInterfaceType>
+      <state_interface>orientation</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>acceleration</state_interface>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:690]:
```suggestion
      <state_interface name="orientation">
          <param name="min">-54</param>
          <param name="max">23</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="acceleration">
          <param name="min">-10</param>
          <param name="max">10</param>
      </state_interface>
```

**Review by destogl — CHANGES_REQUESTED** [review]
Find/Replace All didn't work perfectly in all cases.

Can you also add one more example showing the "user-friendly" joints?

```
1.1 Industrial Robots with only one interface - user-friendly Joints

The `ros2_control_components/PositionJoint` Joint is "position" command and state handles.


  <ros2_control name="RRBotSystemPositionOnly" type="system">
    <hardware>
      <classType>ros2_control_demo_hardware/RRBotSystemPositionOnlyHardware</classType>
      <param name="example_param_write_for_sec">2</param>
      <param name="example_param_read_for_sec">2</param>
    </hardware>
    <joint name="joint1">
      <classType>ros2_control_components/PositionJoint</classType>
      <param name="min">-1</param>
      <param name="max">1</param>
    </joint>
    <joint name="joint2">
      <classType>ros2_control_components/PositionJoint</classType>
      <param name="min">-1</param>
      <param name="max">1</param>
    </joint>
  </ros2_control>

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 15]
```diff
@@ -164,14 +157,18 @@ Note:
...
+        <param name="min">-1</param>
+        <param name="max">1</param>
+      </command_interface>
+      <state_interface name="position"/>
     </joint>
     <joint name="joint2">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:167]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 16]
```diff
@@ -204,14 +193,18 @@ Note:
       <param name="example_param_read_for_sec">2</param>
     </hardware>
     <joint name="joint1">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:196]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 17]
```diff
@@ -204,14 +193,18 @@ Note:
...
+        <param name="min">-1</param>
+        <param name="max">1</param>
+      </command_interface>
+      <state_interface name="position"/>
     </joint>
     <joint name="joint2">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:203]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 18]
```diff
@@ -241,9 +234,11 @@ Note:
       <param name="example_param_read_for_sec">3</param>
     </hardware>
     <joint name="joint1">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:237]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 19]
```diff
@@ -253,9 +248,11 @@ Note:
       <param name="example_param_read_for_sec">3</param>
     </hardware>
     <joint name="joint2">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:251]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 20]
```diff
@@ -336,14 +329,18 @@ Note:
       <param name="example_param_read_for_sec">3</param>
     </hardware>
     <joint name="joint1">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:332]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 21]
```diff
@@ -336,14 +329,18 @@ Note:
...
+        <param name="min">-1</param>
+        <param name="max">1</param>
+      </command_interface>
+      <state_interface name="position"/>
     </joint>
     <joint name="joint2">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:339]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 22]
```diff
@@ -38,25 +38,22 @@ Note:
...
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:48]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 23]
```diff
@@ -70,32 +67,30 @@ Note:
       <param name="example_param_read_for_sec">2</param>
     </hardware>
     <joint name="joint1">
-      <classType>ros2_control_components/MultiInterfaceJoint</classType>
-      <commandInterfaceType name="position">
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:70]:
```suggestion
      <command_interface name="position">
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 24]
```diff
@@ -164,14 +157,18 @@ Note:
       <param name="example_param_read_for_sec">2</param>
     </hardware>
     <joint name="joint1">
-      <classType>ros2_control_components/PositionJoint</classType>
-      <param name="min">-1</param>
-      <param name="max">1</param>
+      <command_interface  name="position">
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:160]:
```suggestion
      <command_interface name="position">
```

**Comment by bmagyar** [comment]
@destogl thanks for the feedback. I've updated your suggestons to not add minmax where they werent originally.

File: design_drafts/components_architecture_and_urdf_examples.md (hunk 14 above)
Comment (bmagyar) [reply design_drafts/components_architecture_and_urdf_examples.md:687]: limits on state interfaces is a discussion we need to have in the new year

**Comment by bmagyar** [comment]
I'd also prefer to not have the simple helpers in here yet. Theres a lot to implement first.  perhaps a new page with extension proposals?

File: design_drafts/components_architecture_and_urdf_examples.md (hunk 14 above)
Comment (destogl) [reply design_drafts/components_architecture_and_urdf_examples.md:687]: I agree.

The long version:

we should leave them as optional arguments. I would add it for now, and what users are doing with it, is their decision. We don't have to provide any handling for them now.

The reason behind it, is that sometimes is good to know the limits of your sensor to know if the data are reliable. E.g. every FTS has calibration range, but the sensors itself can provide data also outside this range (or should this be sensor-specific parameters)

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 25]
```diff
@@ -415,285 +414,281 @@ Note:
...
-      <commandInterfaceType>Kp</commandInterfaceType>
-      <commandInterfaceType>Kd</commandInterfaceType>
-      <commandInterfaceType>effort_limit</commandInterfaceType>
-      <stateInterfaceType>position</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>effort</stateInterfaceType>
-      <stateInterfaceType>coil_resistance</stateInterfaceType>
-      <stateInterfaceType>contact_information</stateInterfaceType>
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      <command_interface name="Kp"/>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:672]: I made a mistake here...
```suggestion
      </command_interface>
      <command_interface name="Kp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 26]
```diff
@@ -415,285 +414,281 @@ Note:
...
-      </commandInterfaceType>
-      <commandInterfaceType>Kp</commandInterfaceType>
-      <commandInterfaceType>Kd</commandInterfaceType>
-      <commandInterfaceType>effort_limit</commandInterfaceType>
-      <stateInterfaceType>position</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>effort</stateInterfaceType>
-      <stateInterfaceType>coil_resistance</stateInterfaceType>
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      <command_interface name="Kp"/>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:495]: I made a mistake here...
```suggestion
      </command_interface>
      <command_interface name="Kp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 27]
```diff
@@ -415,285 +414,281 @@ Note:
...
-      <commandInterfaceType>Kp</commandInterfaceType>
-      <commandInterfaceType>Kd</commandInterfaceType>
-      <commandInterfaceType>effort_limit</commandInterfaceType>
-      <stateInterfaceType>position</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>effort</stateInterfaceType>
-      <stateInterfaceType>coil_resistance</stateInterfaceType>
-      <stateInterfaceType>contact_information</stateInterfaceType>
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      <command_interface name="Kp"/>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:605]: I made a mistake here...
```suggestion
      </command_interface>
      <command_interface name="Kp"/>
```

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 28]
```diff
@@ -415,285 +414,281 @@ Note:
...
-      <commandInterfaceType>Kp</commandInterfaceType>
-      <commandInterfaceType>Kd</commandInterfaceType>
-      <commandInterfaceType>effort_limit</commandInterfaceType>
-      <stateInterfaceType>position</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>effort</stateInterfaceType>
-      <stateInterfaceType>coil_resistance</stateInterfaceType>
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface name="Kp"/>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:429]: Do we need here "min/max" parameters?
I think we defined them as mandatory for the `command_interfaces`...

**Review by destogl — CHANGES_REQUESTED** [review]

File: design_drafts/components_architecture_and_urdf_examples.md (hunk 28 above)
Comment (destogl) [reply design_drafts/components_architecture_and_urdf_examples.md:429]: Ok, they are not mandatory...

File: design_drafts/components_architecture_and_urdf_examples.md (hunk 14 above)
Comment (bmagyar) [reply design_drafts/components_architecture_and_urdf_examples.md:687]: :+1:

**Comment by Karsten1987** [comment]
are we having a follow up ticket to this for updating the parser?

**Comment by Karsten1987** [comment]
Also, shouldn't we rename `classType` to something else? what about just naming it `plugin` similar to what the gazebo tag is using?

**Pull request merged**